### PR TITLE
Add more eslint rules

### PR DESCRIPTION
**Pull request by suneettipirneni**
Adds enforcement for semicolons, single quote strings (where applicable) and errors for unused variables.

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -14,8 +14,16 @@
...
     "prefer-const": "error",
+    // note you must disable the base rule as it can report incorrect errors
+    "semi": "off",
+    "@typescript-eslint/semi": ["error"],
     "space-before-blocks": "error",
     "spaced-comment": "warn",
+    "quotes": "off",
+    "@typescript-eslint/quotes": ["error", "single", { "avoidEscape": true }],
+    // note you must disable the base rule as it can report incorrect errors
+    "no-unused-vars": "off",
```
Comment (tanndlin) [inline .eslintrc.json:25]: That's odd, bc i swear we had this rule before, but i dont see it in the current eslint.json

**Review by tanndlin — APPROVED** [review]
I like this because we had prettier rules for this, but sometimes a new contributor doesnt have prettier.

File: src/test.ts [hunk 2]
```diff
@@ -1,7 +0,0 @@
-class MyClass {
-  private _myVar = 'hello';
-
-  foo() {
-    console.log(this._myVar)
-  }
-}
```
Comment (tanndlin) [inline src/test.ts:7]: Whats this about?

**Review by tanndlin — APPROVED** [review]

File: src/test.ts (hunk 2 above)
Comment (suneettipirneni) [reply src/test.ts:7]: It was some leftover testing code that leaked through

**Pull request merged**